docs/docs/module_guides/indexing/lpg_index_guide.md

```diff
@@ -0,0 +1,573 @@
+
+Here's a simple (but naive) example:
+
+```python
+prompt = (
+    "Some text is provided below. Given the text, extract up to "
+    "{max_paths_per_chunk} "
+    "knowledge triples in the form of `subject,predicate,object` on each line. Avoid stopwords.\n"
+)
+
+
+def parse_fn(response_str: str) -> List[Tuple[str, str, str]]:
+    lines = response_str.split("\n")
+    triples = [line.split(",") for line in lines]
+    return triples
+
+
+kg_extractor = SimpleLLMPathExtractor(
+    llm=llm,
+    extract_prompt=prompt,
+    parse_fn=parse_fn,
+)
+```
+
+#### (default) `ImplicitPathExtractor`
+
+Extract paths using the `node.relationships` attribute on each llama-index node object.
+
+This extractor does not need an LLM or embedding model to run, since it's merely parsing properties that already exist on llama-index node objects.
+
+```python
+from llama_index.core.indices.property_graph import ImplicitPathExtractor
+
+kg_extractor = ImplicitPathExtractor()
+```
+
+#### `SchemaLLMPathExtractor`
+
+Extract paths following a strict schema of allowed entities, relationships, and which entities can be connected to which relationships.
+
+Using pydantic, structured outputs from LLMs, and some clever validation, we can dynamically specify a schema and verify the extractions per-path.
+
+```python
+from typing import Literal
+from llama_index.core.indices.property_graph import SchemaLLMPathExtractor
+
+# recommended uppercase, underscore separated
+entities = Literal["PERSON", "PLACE", "THING"]
+relations = Literal["PART_OF", "HAS", "IS_A"]
+schema = {
+    "PERSON": ["PART_OF", "HAS", "IS_A"],
+    "PLACE": ["PART_OF", "HAS"],
+    "THING": ["IS_A"],
+}
+
+kg_extractor = SchemaLLMPathExtractor(
+    llm=llm,
+    possible_entities=entities,
+    possible_relations=relations,
+    kg_validation_schema=schema,
+    strict=True,  # if false, will allow triples outside of the schema
+    num_workers=4,
+    max_paths_per_chunk=10,
+    show_progres=False,
+)
+```
+
+This extractor is extremely customizable, and has options to customize
+- various aspects of the schema (as seen above)
+- the `extract_prompt`
+- `strict=False` vs. `strict=True`, to allow triples outside of the schema or not
+- passing in your own custom `kg_schema_cls` if you are a pydantic pro and wanted to create you own pydantic class with custom validation.
+
+### Retrieval and Querying
+
+Labeled property graphs can be queried in several ways to retrieve nodes and paths. And in LlamaIndex, we can combine several node retrieval methods at once!
+
+```python
+# create a retriever
+retriever = index.as_retriever(sub_retrievers=[retriever1, retriever2, ...])
+
+# create a query engine
+query_engine = index.as_query_engine(
+    sub_retrievers=[retriever1, retriever2, ...]
+)
+```
+
+If no sub-retrievers are provided, the defaults are
+`LLMSynonymRetriever` and `VectorContextRetriever` (if embeddings are enabled).
+
+All retrievers currently include:
```

Comment:
ubernit: include `PGRetriever` (if user wants to define class explicitly instead of `index.as_retriever`)

Reply:
Yea for sure. That kind of deserves its own little section, since its meant to drive all these sub retrievers